src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -486,11 +511,9 @@ void runtime::gpu::GPU_ExternalFunction::emit_temp_mem_pool_allocation(
     }
     if (m_temporaries_used)
     {
-        size_t temp_pool_size = current_function->get_temporary_pool_size();
         m_writer << "// Allocate the memory pool\n";
-        // TODO memory pool malloc.
-        m_writer << "void* pool_base_ptr = ngraph::runtime::gpu::create_gpu_buffer("
-                 << temp_pool_size << ");\n";
+        m_writer << "void* pool_base_ptr = ngraph::runtime::gpu::invoke_memory_primitive(ctx, "
```

Comment:
```
+        // reserve memory buffers for intermediate results
 +        size_t temp_pool_size = current_function->get_temporary_pool_size();
 +        if (temp_pool_size)
 +        {
 +            size_t pool_idx = allocator.reserve_workspace(temp_pool_size, false);
 +            m_memory_buffers[current_function->get_name()] = pool_idx;
 +        }
```
should be here?

Reply:
like this:
```
 // reserve memory buffers for intermediate results
  size_t temp_pool_size = current_function->get_temporary_pool_size();
  size_t pool_idx = allocator.reserve_workspace(temp_pool_size, false);
  m_writer << "void* pool_base_ptr = ngraph::runtime::gpu::invoke_memory_primitive(ctx," <<  pool_idx << ");\n"
```

Reply:
I think this is the comment as below. Since the common function elimination is calling the emitters twice, these had to be separated so that the allocation reservations are made prior to the emitters being called.

If @rkimballn1 comes up with a solution for the CFE pass to only call the emitters once (avoiding the double allocations - until we formally fix that in #1246 - then we can do as you suggest and put the reservations into `emit_temp_mem_pool_allocation`.

Reply:
Make sense.